public/electron.js

```diff
@@ -339,9 +339,14 @@ app.on('will-quit', () => {
     stream = false;
 
     // copy file to config location
-    fs.mkdir(savePath, { recursive: true }, (err) => {
-      log.error(err);
-      fs.copyFileSync(preSavePath, getFullPath(`pid_${participantID}_${today.getTime()}.json`));
-    });
+    const fullPath = getFullPath(`pid_${participantID}_${today.getTime()}.json`);
+    try {
+      fs.mkdirSync(savePath, { recursive: true });
+      fs.copyFileSync(preSavePath, fullPath);
+    } catch (e) {
+      console.error('Unable to save file: ', fullPath);
+      console.error(e);
+      log.error(e);
```

Comment:
what does log.error do that console.error doesn't? might need to add to my toolbelt!

Reply:
It's a special logging tool needed by electron. https://www.npmjs.com/package/electron-log